### ISSUE TITLE
logging.pattern.console override no longer works

### ISSUE THREAD

**Issue by bguedel** [spring-cloud/spring-cloud-sleuth#1893]
**Describe the bug**
> Please provide details of the problem, including the version of Spring Cloud that you
are using. 

In spring 2.3.x we set a custom logging pattern due to a line break issue. Normally we just use the default spring logging pattern, with the custom line break code. Sleuth 2.x was cool with this, and just overwrites `LOG_LEVEL_PATTERN`

Moving to spring 2.4.4 sleuth is updated to 3.x and the pattern no longer works.

**Sample**
> If possible, please provide a test case or a minimal sample **Java** application that reproduces
the problem. This makes it much easier for us to diagnose the problem and to verify that
we have fixed it.

Spring Boot version 2.4.4
Spring Cloud Dependencies 2020.0.2
  org.springframework.cloud:spring-cloud-starter-sleuth -> 3.0.2

```yaml
logging:
  level:
    root: WARN
  pattern:
    console: "${CONSOLE_LOG_PATTERN:%clr(%d{${LOG_DATEFORMAT_PATTERN:yyyy-MM-dd HH:mm:ss.SSS}}){faint} %clr(${LOG_LEVEL_PATTERN:%5p}) %clr(${PID: }){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%replace(%xException){'\n','\u2028'}%nopex%n}"
```

Removing the console pattern resolves it back to working.

**Comment by jonatan-ivanov** [comment]
I'm not sure if your pattern is correct, in logback config the default value comes after `:-` not `:`, could you please try out what happens if you put [this](https://github.com/spring-projects/spring-boot/blob/72a1eb6384bc5ca9171e0433dc6a3bb750c8e8b0/spring-boot-project/spring-boot/src/main/resources/org/springframework/boot/logging/logback/defaults.xml#L12) into your `logback-spring.xml` as the [documentation suggests](https://docs.spring.io/spring-cloud-sleuth/docs/current/reference/htmlsingle/#features-log-integration-json-logback)?

**Comment by marcingrzejszczak** [comment]
In which file do you set this configuration?

**Comment by bguedel** [comment]
src/main/resources/application.yml
```yml
spring:
  application:
    name: test-server

logging:
  pattern:
    console: "${CONSOLE_LOG_PATTERN:%clr(%d{${LOG_DATEFORMAT_PATTERN:yyyy-MM-dd HH:mm:ss.SSS}}){faint} %clr(${LOG_LEVEL_PATTERN:%5p}) %clr(${PID: }){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%replace(%xException){'\n','\u2028'}%nopex%n}"
```
output
```sh
2021-04-02 08:54:30.803  INFO 6996 --- [  restartedMain] c.d.bguedel.test.TestServerApplication   : Started TestServerApplication in 4.011 seconds (JVM running for 4.444)
```
---
```yml
spring:
  application:
    name: test-server

# logging:
#   pattern:
#     console: "${CONSOLE_LOG_PATTERN:%clr(%d{${LOG_DATEFORMAT_PATTERN:yyyy-MM-dd HH:mm:ss.SSS}}){faint} %clr(${LOG_LEVEL_PATTERN:%5p}) %clr(${PID: }){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%replace(%xException){'\n','\u2028'}%nopex%n}"
```

output
```sh
2021-04-02 08:54:48.992  INFO [test-server,,] 7263 --- [  restartedMain] c.d.bguedel.test.TestServerApplication   : Started TestServerApplication in 4.278 seconds (JVM running for 4.806)
```

**Comment by bguedel** [comment]
```yml
spring:
  application:
    name: test-server

logging:
  pattern:
    console: "${CONSOLE_LOG_PATTERN:-%clr(%d{${LOG_DATEFORMAT_PATTERN:-yyyy-MM-dd HH:mm:ss.SSS}}){faint} %clr(${LOG_LEVEL_PATTERN:-%5p}) %clr(${PID:- }){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%replace(%xException){'\n','\u2028'}%nopex%n}"
```

output
```sh
--2021-04-02 09:03:01.635 - INFO 8028 --- [  restartedMain] c.d.bguedel.test.TestServerApplication   : Started TestServerApplication in 4.164 seconds (JVM running for 4.632)
```

**Comment by marcingrzejszczak** [comment]
Could you please set up a sample that replicates the issue?

**Comment by bguedel** [comment]
i think if you just grab a default start.spring.io build it should show it. let me create one.

**Comment by bguedel** [comment]
https://github.com/bguedel/demo

Logging issue
```sh
SPRING_PROFILES_ACTIVE=bad_logging ./gradlew bootrun
```

No Issue
```sh
./gradlew bootrun
```

**Comment by bguedel** [comment]
if you change the build.gradle

```groovy
id 'org.springframework.boot' version '2.3.9.RELEASE'
...
	set('springCloudVersion', "Hoxton.SR10")
```

and run the bad_logging, it seems the applicaiton name doesnt show up, but the sleuth section does show up.

**Comment by jonatan-ivanov** [comment]
Thanks for the sample and the details, I was able to repro it.

### With Hoxton

You needed to put `spring.application.name` into the `bootstrap.yml` (this has changed in `2020.x`).
If I set `spring.application.name` in `bootstrap.yml` and  `logging.pattern.console` in `application.xml` it works.

### With Ilford (`2020.x`)

If you remove `CONSOLE_LOG_PATTERN` from your property, it works. I don't know why, I'm going to investigate this further, just wanted to give you a quick workaround:

```
logging.pattern.console: "%clr(%d{${LOG_DATEFORMAT_PATTERN:yyyy-MM-dd HH:mm:ss.SSS}}){faint} %clr(${LOG_LEVEL_PATTERN:%5p}) %clr(${PID: }){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%n${LOG_EXCEPTION_CONVERSION_WORD:%wEx}"
```

**Comment by jonatan-ivanov** [comment]
Might be because [as the docs says](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-custom-log-configuration) `logging.pattern.console` and `CONSOLE_LOG_PATTERN` are the same property so I'm not sure if it makes sense to do `logging.pattern.console:${CONSOLE_LOG_PATTERN:...}` so I'm not sure anymore hat this is really a bug though something definitely happened under the hood that changes the behavior.

**Comment by bguedel** [comment]
thanks for the response.
I thought I had tested without the CONSOLE_LOG_PATTERN, but I must have missed that.

I can confirm that putting your ^^ `logging.pattern.console` in my demo app, both the app name and the sleuth section do show up.

I agree, I am not sure this is a bug as much as this might have been an accidental feature in the previous version, that has been corrected in this version.

x = x or 3  <--- this doesnt make sense
